mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb

```diff
@@ -4847,11 +4832,11 @@ func expandNotificationConfig(configured interface{}) *container.NotificationCon
 	}
 }
 
-func expandBinaryAuthorization(configured interface{}) *container.BinaryAuthorization {
+func expandBinaryAuthorization(configured interface{}, legacy_enabled bool) *container.BinaryAuthorization {
```

Comment:
Based on [your comment](https://github.com/hashicorp/terraform-provider-google/issues/16759#issuecomment-1852836055) on the ticket, it sounds like the behavior of the expander _might_ already be correct as-is - it's setting / sending `Enabled: config["enabled"]`, which should be false if the value isn't set, which apparently is fine? But the issue is maybe sending `true` for `enabled` while also setting a policy mode?

What exactly is the failure case we're addressing? Do we want to have the resource return a validation error if `enabled = true` and policy mode is also set? (That would require a customize diff func.)

Reply:
Assuming there is only a diff in the `binary_authorization` field, the expander behavior is correct. The case I was addressing was if there was a diff in `binary_authorization` and `enable_binary_authorization`. It sounds like we can assume the latter will never have a diff, so we can leave this as is.

> Do we want to have the resource return a validation error if enabled = true and policy mode is also set?

I think we can just have the backend do the validation in this case.